### PR TITLE
refs #3021 implemented Program::issueModuleCmd()

### PR DESCRIPTION
**Pull request by davidnich**

### CODE REVIEW

**Comment by davidnich** [comment]
@sejvlond please wait with this

**Comment by davidnich** [comment]
@sejvlond now it's OK

File: lib/ModuleManager.cpp [hunk 1]
```diff
@@ -1414,7 +1418,27 @@ void QoreModuleManager::issueParseCmd(const QoreProgramLocation* loc, const char
...
+    }
+
+    QoreAbstractModule* mi = findModule(mname);
+    assert(mi);
+
+    // ensure the program is in context
+    QoreProgramContextHelper pch(pgm);
+    mi->issueModuleCmd(&loc_builtin, cmd, xsink);
+
+    // enrich exception deescription if present
```
Comment (sejvlond) [inline lib/ModuleManager.cpp]: deescription typo

File: doxygen/lang/900_release_notes.dox.tmpl [hunk 2]
```diff
@@ -121,6 +128,7 @@
       - <a href="../../modules/Swagger/html/index.html">Swagger</a>:
         - Swagger module does not accept multipart/form-data content-type and also does not work with list and hash parameters
           (<a href="https://github.com/qorelanguage/qore/issues/2932">issue 2932</a>)
+<<<<<<< Updated upstream
```
Comment (sejvlond) [inline doxygen/lang/900_release_notes.dox.tmpl]: <<<<<<< left from merging

File: doxygen/lang/900_release_notes.dox.tmpl [hunk 3]
```diff
@@ -132,6 +140,19 @@
         - <a href="../../modules/DebugHandler/html/index.html">DebugHandler</a> reimplemented to support multiple websocket handlers
         - program is not interrupted at bootstrap code
         - command line utils display source line code when interrupted
+=======
```
Comment (sejvlond) [inline doxygen/lang/900_release_notes.dox.tmpl]: ======= left from merging

File: doxygen/lang/900_release_notes.dox.tmpl [hunk 4]
```diff
@@ -132,6 +140,19 @@
...
+      - Fractional date components are accepted in the @ref short_reldates based on <a href="https://en.wikipedia.org/wiki/ISO_8601#Durations">ISO-8601 durations</a>
+    - Improved debugging support:
+      - @ref Qore::ProgramControl::getStatementIdInfo() "ProgramControl::getStatementIdInfo()" provides breakpoint info
+      - Command to resolve statement
+      - Interrupt notification provides call stack info
+      - \c "Run to statement" implementation
+      - <a href="../../modules/DebugHandler/html/index.html">DebugHandler</a> reimplemented to support multiple websocket handlers
+      - Programs are not interrupted in bootstrap code
+      - Command line utils display source line code when interrupted
+>>>>>>> Stashed changes
```
Comment (sejvlond) [inline doxygen/lang/900_release_notes.dox.tmpl]: >>>>>>>  left from merging

File: include/qore/intern/ModuleInfo.h [hunk 5]
```diff
@@ -611,7 +614,7 @@ class QoreBuiltinModule : public QoreAbstractModule {
         return dlptr;
     }
 
-    DLLLOCAL virtual void issueParseCmd(const QoreProgramLocation* loc, QoreString &cmd);
+    DLLLOCAL virtual void issueModuleCmd(const QoreProgramLocation* loc, const QoreString& cmd, ExceptionSink* xsink);
```
Comment (sejvlond) [inline include/qore/intern/ModuleInfo.h]: we should use `override` here

File: lib/QC_Program.qpp [hunk 6]
```diff
@@ -2013,3 +2013,32 @@ list<int> Program::getThreadList() [flags=CONSTANT] {
...
+        return QoreValue();
+    }
+
+    TempEncodingHelper tmp_cmd(cmd, QCS_DEFAULT, xsink);
+    if (!tmp_cmd) {
+        return QoreValue();
+    }
+
+    QMM.issueRuntimeCmd(tmp_module->c_str(), p, **tmp_cmd, xsink);
+}
```
Comment (sejvlond) [inline lib/QC_Program.qpp]: no newline at the end of the file - I don't know if it's an issue, I just find out that github is showing a special char for this case and we probably always have a newline at the end...

**Review by sejvlond — CHANGES_REQUESTED** [review]

File: doxygen/lang/900_release_notes.dox.tmpl (hunk 2 above)
Comment (davidnich) [reply doxygen/lang/900_release_notes.dox.tmpl]: fixed

File: doxygen/lang/900_release_notes.dox.tmpl (hunk 3 above)
Comment (davidnich) [reply doxygen/lang/900_release_notes.dox.tmpl]: fixed

File: include/qore/intern/ModuleInfo.h (hunk 5 above)
Comment (davidnich) [reply include/qore/intern/ModuleInfo.h]: fixed

File: doxygen/lang/900_release_notes.dox.tmpl (hunk 4 above)
Comment (davidnich) [reply doxygen/lang/900_release_notes.dox.tmpl]: fixed

File: lib/QC_Program.qpp (hunk 6 above)
Comment (davidnich) [reply lib/QC_Program.qpp]: fixed

**Review by sejvlond — APPROVED** [review]

**Pull request merged**